File: server/src/Bananagrams.Dal/Models/GameAnagram.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Bananagrams.Dal.Models;

[Table("game_anagrams")]
public class GameAnagram
{
    [Key]
    [Column("id")]
    public int Id { get; set; }
    [Column("anagram_word")]
    public string AnagramWord { get; set; }
    [Column("date_created")]
    public DateTime DateCreated { get; set; }
    [Column("order_sequence")]
    public int Order { get; set; }
    [Column("game_id")]
    public int GameId { get; set; }
    [Column("game_anagram_type_id")]
    public int GameAnagramTypeId { get; set; }
    [Column("word_id")]
    public int WordId { get; set; }
    
    public Game? Game { get; set; }
    public GameAnagramType? GameAnagramType { get; set; }
    public Word? Word { get; set; }
    public List<GameUserGameAnagram>? GameUserGameAnagrams { get; set; }
}

File: server.java/bananagrams/src/main/java/coe/bananagrams/entity/GameAnagram.java
package coe.bananagrams.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "game_anagrams")
public class GameAnagram {
    @Id
    @Column(name = "id", updatable = false, nullable = false)
    private int id;
    @Column(name = "anagram_word", nullable = false)
    private String anagramWord;
    @Column(name = "date_created", nullable = false)
    private Date dateCreated;
    @Column(name = "order_sequence", nullable = false)
    private int order;
    @Column(name = "game_id", insertable = false, updatable = false, nullable = false)
    private int gameId;
    @Column(name = "game_anagram_type_id", insertable = false, updatable = false, nullable = false)
    private int gameAnagramTypeId;
    @Column(name = "word_id", insertable = false, updatable = false, nullable = false)
    private int wordId;

    @ManyToOne()
    private Game game;
    @ManyToOne()
    private GameAnagramType gameAnagramType;
    @ManyToOne()
    private Word word;
    @OneToMany(mappedBy = "gameAnagram")
    private List<GameUserGameAnagram> gameUserGameAnagrams;
}
